docs: document testing (#127)
kyma-project/kyma-companion: tests/blackbox/evaluation.md

For the evaluation, we need to define a set of scenarios, with each scenario further divided into expectations. Kyma Companion is prompted through its endpoints to evaluate the given scenario. In the Evaluation, the response is compared to the given expectations. This comparison will either result in a match or no match. However, as different expectations can be more or less complex, this boolean value is multiplied by a complexity factor. Every expectation must be evaluated multiple times so the idempotency performance of Kyma Companion can be calculated.

## Setup

The test cases are deployed on a Gardener cluster. The cluster is set up by hand and is not deleted unless necessary.

> Once the cluster is set up, the [Setup Gardener Test Cluster](https://github.com/kyma-project/kyma-companion/actions/workflows/setup-test-cluster.yaml) GitHub Action can be manually triggered.

Reviewer: ```suggestion
When the cluster is set up, manually trigger the GitHub Action [Setup Gardener Test Cluster](https://github.com/kyma-project/kyma-companion/actions/workflows/setup-test-cluster.yaml).
```

The action automatically connects to the Gardener cluster, cleans it and deploys all the examples in the [tests/data/evaluation/namespace-scoped](../../tests/data/evaluation/namespace-scoped) directory.

> The GitHub Action provides a summary of any missing scenarios or ones that failed to be deployed. These need to be fixed within their own `deploy.sh` script.

Reviewer: ```suggestion
The GitHub Action provides a summary of any missing scenarios or ones that failed to be deployed. You must fix these within their own `deploy.sh` script.
```
